### ISSUE TITLE
[Translation tool] lt_LT

### ISSUE THREAD

**Issue by Gamesh** [briannesbitt/Carbon#1792]
<details><summary>Original report</summary>
<p>

|English|lt_LT|comments|
|--|--|--|
|2 years|2 metus|2 metai|
|2yrs|2 metus|2 m.|
|1 year|1 metus|1 metai|
|a year|1 metus|metai|
|2 months|2 mėnesius|2 mėnesiai|
|2mos|2 mėnesius|2 mėn.|
|1 month|1 mėnesį|1 mėnuo|
|a month|1 mėnesį|mėnuo|
|2 weeks|2 savaites|2 savaitės|
|2w|2 savaites|2 sav.|
|1 week|1 savaitę|1 savaitė|
|a week|1 savaitę|savaitė|
|2 days|2 dienas|2 dienos|
|2d|2 dienas|2 d.|
|1 day|1 dieną|1 diena|
|a day|1 dieną|diena|
|2 hours|2 valandas|2 valandos|
|2h|2 valandas|2 val.|
|1 hour|1 valandą|1 valanda|
|an hour|1 valandą|valanda|
|2 minutes|2 minutes|2 minutės|
|2m|2 minutes|2 min.|
|1 minute|1 minutę|1 minutė|
|a minute|1 minutę|1 min.|
|2 seconds|2 sekundes|2 sekundės|
|2s|2 sekundes|2 sek.|
|1 second|1 sekundę|1 sekundė|
|a few seconds|1 sekundę|kelios sekundės|
|Monday|Pirmadienis|pirmadienis|
|Tuesday|Antradienis|antradienis|
|Wednesday|Trečiadienis|Trečiadienis|
|Thursday|Ketvirtadienis|ketvirtadienis|
|Friday|Penktadienis|penktadienis|
|Saturday|Šeštadienis|šeštadienis|
|Sunday|Sekmadienis|sekmadienis|
|January|Sausis|sausis|
|February|Vasaris|vasaris|
|March|Kovas|kovas|
|April|Balandis|balandis|
|May|Gegužė|gegužė|
|June|Birželis|birželis|
|July|Liepa|liepa|
|August|Rugpjūtis|rugpjūtis|
|September|Rugsėjis|rugsėjis|
|October|Spalis|spalis|
|November|Lapkritis|lapkritis|
|December|Gruodis|gruodis|
|2 hours before|2 valandas nuo dabar|prieš 2 valandas|
|2 hours after|po 2 valandas|po 2 valandų|

</p>
</details>

Changes addressed in 2.21:

|English|lt_LT|comments|
|--|--|--|
|2yrs|2 metus|2 m.|
|2mos|2 mėnesius|2 mėn.|
|2w|2 savaites|2 sav.|
|2d|2 dienas|2 d.|
|2h|2 valandas|2 val.|
|2m|2 minutes|2 min.|
|a minute|1 minutę|1 min.|
|2s|2 sekundes|2 sek.|
|Monday|Pirmadienis|pirmadienis|
|Tuesday|Antradienis|antradienis|
|Wednesday|Trečiadienis|Trečiadienis|
|Thursday|Ketvirtadienis|ketvirtadienis|
|Friday|Penktadienis|penktadienis|
|Saturday|Šeštadienis|šeštadienis|
|Sunday|Sekmadienis|sekmadienis|
|January|Sausis|sausis|
|February|Vasaris|vasaris|
|March|Kovas|kovas|
|April|Balandis|balandis|
|May|Gegužė|gegužė|
|June|Birželis|birželis|
|July|Liepa|liepa|
|August|Rugpjūtis|rugpjūtis|
|September|Rugsėjis|rugsėjis|
|October|Spalis|spalis|
|November|Lapkritis|lapkritis|
|December|Gruodis|gruodis|

Other changes are postponed for now, see #1800

**Comment by kylekatarnls** [comment]
Thank you very much. I will check this soon.

**Comment by Gamesh** [comment]
@kylekatarnls thanks, if you have any questions let me know

**Comment by kylekatarnls** [comment]
I see there are declination metai/metus savaitę/savaitė, can you explain when to use which ones?

**Comment by Gamesh** [comment]
well we have these cases in Lithuanian but bare with me :)
so when you use it in a sentence describing what or whom this is accusative: 
```
you have 2 years 
tu  turi 2 metus
```
when you use it in general you say "2 years" -> "2 metai" that's nominative
https://en.wikipedia.org/wiki/Grammatical_case#Indo-European_languages

the same goes for the week.

if i print duration lets say 
`1 year 2 weeks 1 day`
it should be in nominative:
`1 metai 2 savaitės 1 diena`

there is also a slight difference for number above 9
`10 years` -> `10 metų`
`10 weeks` -> `10 savaičių`

**Comment by kylekatarnls** [comment]
I will need the exact rules and declination.

The file is: https://github.com/briannesbitt/Carbon/blob/master/src/Carbon/Lang/lt.php

By default we should have nominative in unit translations, but if we do so, we need to do transformations for after/before/from_now/ago keys:

```php
'before' => function ($time) {
  $time = strtr($time, [
    'valanda' => 'valandas',
    // ...
  ]);

  return 'prieš '.$time;
},
```

**Comment by Gamesh** [comment]
i see we have some logic here
`'year' => ':count metus|:count metus|:count metų',`
should it be like 1st part nominative singular, 2nd - nominative plural, 3rd when more than 10?

**Comment by kylekatarnls** [comment]
`|` is the plurialization notation, for `lt` locale, the rule is:
`((1 == $number % 10) && (11 != $number % 100)) ? 0 : ((($number % 10 >= 2) && (($number % 100 < 10) || ($number % 100 >= 20))) ? 1 : 2)`

It means with `'a|b|c'`:
If ends with 1 but do not end with 11, take `a`
Else if ends with 03, 04, 05, 06, 07, 08 or 09 or ends with 2x, 3x, 4x, 5x, 6x, 7x, 8x or 9x, take `b`
Else, take `c`

**Comment by kylekatarnls** [comment]
I partially fixed Lithuanian translations. Lower case and short units are implemented to be released in 2.21.1. See #1800 for next changes.
